mmdet/core/bbox/assigners/approx_max_iou_assigner.py

```diff
@@ -81,11 +81,15 @@ def assign(self,
         Returns:
             :obj:`AssignResult`: The assign result.
         """
-
-        if squares.shape[0] == 0 or gt_bboxes.shape[0] == 0:
-            raise ValueError('No gt or approxs')
         num_squares = squares.size(0)
         num_gts = gt_bboxes.size(0)
+
+        if num_squares == 0 or num_gts == 0:
+            # No predictions and/or truth, return empty assignment
+            overlaps = approxs.new(num_gts, num_squares)
```

Comment:
`overlaps` initialization is not consistent.
In approx_max_iou_assigner: `overlaps = approxs.new(num_gts, num_squares)`
In max_iou_assigner: `max_overlaps = overlaps.new_zeros((num_bboxes, ))`
In point_assigner: `max_overlaps = None`